pkg/handler/collector/collector_test.go

```diff
@@ -80,3 +88,108 @@ func TestCollect(t *testing.T) {
 		})
 	}
 }
+
+func Test_Publish(t *testing.T) {
+	expectedDocTree := dochelper.DocNode(&testdata.Ite6SLSADoc)
+
+	natsTest := nats_test.NewNatsTestServer()
+	url, err := natsTest.EnableJetStreamForTest()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer natsTest.Shutdown()
+
+	ctx := context.Background()
+	jetStream := emitter.NewJetStream(url, "", "")
+	ctx, err = jetStream.JetStreamInit(ctx)
+	if err != nil {
+		t.Fatalf("unexpected error initializing jetstream: %v", err)
+	}
+	err = jetStream.RecreateStream(ctx)
+	if err != nil {
+		t.Fatalf("unexpected error recreating jetstream: %v", err)
+	}
+	defer jetStream.Close()
+	err = Publish(ctx, &testdata.Ite6SLSADoc)
+	if err != nil {
+		t.Fatalf("unexpected error on emit: %v", err)
+	}
+
+	var cancel context.CancelFunc
+
+	ctx, cancel = context.WithTimeout(ctx, 2*time.Second)
+	defer cancel()
+
+	docChan := make(chan processor.DocumentTree, 1)
+	errChan := make(chan error, 1)
+	defer close(errChan)
+	go func() {
+		errChan <- testSubscribe(ctx, docChan)
+	}()
+
+	numSubscribers := 1
+	subscribersDone := 0
+
+	for subscribersDone < numSubscribers {
+		select {
+		case d := <-docChan:
+			if !dochelper.DocTreeEqual(d, expectedDocTree) {
+				t.Errorf("doc tree did not match up, got\n%s, \nexpected\n%s", dochelper.StringTree(d), dochelper.StringTree(expectedDocTree))
+			}
+		case err := <-errChan:
+			if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+				t.Errorf("nats emitter Subscribe test errored = %v", err)
+			}
+			subscribersDone += 1
+		}
+	}
+
```

Comment:
need to drain the channel here as well, since the loop may end without deref'ing docChan